samcli/lib/build/bundler.py

```diff
@@ -134,6 +142,61 @@ def set_sourcemap_env_from_metadata(self) -> Dict:
 
         return template
 
+    def _should_update_handler(self, handler: str, name: str) -> bool:
+        """
+        Function to check if the handler exists in the build dir where we expect it to.
+        If it does, we won't change the path to prevent introducing breaking changes.
+
+        :param handler: handler string as defined in the template.
+        :param name: function name corresponding to function build directory
+        :return: True if it's an invalid handler, False otherwise
+        """
+        if not self._build_dir:
+            return False
+        try:
+            handler_filename = self._get_path_and_filename_from_handler(handler)
+        except InvalidHandlerPathError:
+            LOG.debug("Unable to parse handler, continuing without post-processing template.")
+            return False
+        expected_artifact_path = Path(self._build_dir, name, handler_filename)
+        return not expected_artifact_path.is_file()
+
+    @staticmethod
+    def _get_path_and_filename_from_handler(handler: str) -> str:
+        """
+        Takes a string representation of the handler defined in the
+        template, returns the file name and location of the handler.
+
+        :param handler: string representation of handler property
+        :return: string path to built handler file
+        """
+        try:
+            path = str(Path(handler).parent / Path(handler).stem) + ".js"
+        except (AttributeError, TypeError):
+            raise InvalidHandlerPathError("Unable to parse the handler")
```

Comment:
If this exception won't be returned back to user (I saw `_should_update_handler` catches this exception and returns `False`), I would instead suggest to return `None` here. 

It will be easier for method user to see method returns `Optional[str]` and take an action (with exceptions in python, it is unclear if a method will raise an exception or not). It will be also easier (and more performant) since we don't need to have 2 times of try/except block.

Reply:
Good call-out, updated.